src/hw10/FileNavigator.java

```diff
@@ -0,0 +1,36 @@
+package hw10;
+
+import java.io.File;
+import java.util.*;
+import java.util.function.BiFunction;
+
+public class FileNavigator extends FileData{
+    static BiFunction<String, String, String> bFunc = (oldValue, newValue)->oldValue+";"+newValue;
+
+    static Map<String, String> listName = new HashMap<>();
+    public static void main(String[] args) {
+        add(path, name);
+        add(path1, "FileData.java");
+        add(path1, "FileNavigator.java");
+        System.out.println("listName = " + listName);
+        System.out.println(find(path));
+        System.out.println(find(path).length());
+        listName.entrySet().stream()
+                .sorted(Map.Entry.<String, String>comparingByValue().reversed())
+                .forEach(System.out::println);
+        remove(path1);
+        System.out.println("listName = " + listName);
+
+    }
+    private static void add(String path, String name){
+
+        listName.merge(path, name, bFunc);
```

Comment:
Cool approach) but the files (a list of instances of FileData) should be in the map